numba/cuda/cudadrv/devicearray.py

```diff
@@ -895,6 +895,8 @@ def check_array_compatibility(ary1, ary2):
     if ary1sq.shape != ary2sq.shape:
         raise ValueError('incompatible shape: %s vs. %s' %
                          (ary1.shape, ary2.shape))
-    if ary1sq.strides != ary2sq.strides:
+    # We check strides only if the size is nonzero, because strides are
+    # irrelevant (and can differ) for zero-length copies.
+    if ary1.size and ary1sq.strides != ary2sq.strides:
```

Comment:
NOTE: cases like:
```python
ary1 = np.ndarray(shape=(1, 1, 1), strides=(0, 2, 4), dtype=np.float64)
ary2 = np.ndarray(shape=(), strides=(), dtype=np.float64)
```
are "safe" due to the `squeeze` above. It shouldn't be possible to reach here and need to check `ary2.size`.

Reply:
Also checked manually with:

```python
from numba import cuda
import numpy as np

x  = np.ndarray(shape = (1, 1, 1), strides = (0, 2, 4), dtype=np.float64)
y  = np.ndarray(shape = (), strides = (), dtype=np.float64)

print(x.shape, x.strides, x.size)
print(y.shape, y.strides, y.size)

dx = cuda.to_device(x)
dy = cuda.to_device(y)

dx.copy_to_host(y)
dy.copy_to_host(x)
```

without issue.

Reply:
Great, thanks for checking this too!